packages/headless/src/controllers/core/search-parameter-manager/headless-core-search-parameter-manager.ts

```diff
@@ -124,20 +125,21 @@ export function validateParams(
 export function getCoreActiveSearchParameters(
   engine: CoreEngine
 ): SearchParameters {
-  const state = engine.state;
+  const {state, logger} = engine;
   return {
-    ...getQ(state),
-    ...getTab(state),
-    ...getSortCriteria(state),
-    ...getFacets(state),
-    ...getCategoryFacets(state),
-    ...getNumericFacets(state),
-    ...getDateFacets(state),
+    ...getQ(state, logger),
+    ...getTab(state, logger),
+    ...getSortCriteria(state, logger),
+    ...getFacets(state, logger),
+    ...getCategoryFacets(state, logger),
+    ...getNumericFacets(state, logger),
+    ...getDateFacets(state, logger),
   };
 }
 
-function getQ(state: Partial<SearchParametersState>) {
+function getQ(state: Partial<SearchParametersState>, logger: Logger) {
   if (state.query === undefined) {
```

Comment:
If the key exists in the state, even if its value is `undefined`, it could be sign that the value was updated to `undefined`.

I think we should check instead whether the key exists in the state like so:
```suggestion
  if ('query' in state) {
```